### PR TITLE
ci: add typescript-eslint rules

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

### CODE REVIEW

File: projects/cdk/abstract/dialog.service.ts [hunk 1]
```diff
@@ -22,10 +22,13 @@ export abstract class AbstractTuiDialogService<T, K = void> extends Observable<
     }
 
     open<G = void>(
+        // eslint-disable-next-line @typescript-eslint/no-invalid-void-type
```
Comment (waterplea) [inline projects/cdk/abstract/dialog.service.ts:25]: Can you elaborate on that?

File: projects/cdk/decorators/test/pure.spec.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-confusing-void-expression */
```
Comment (waterplea) [inline projects/cdk/decorators/test/pure.spec.ts:1]: And that

File: projects/cdk/abstract/dialog.service.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/abstract/dialog.service.ts:25]: Bug of typescript-eslint
https://github.com/typescript-eslint/typescript-eslint/issues/6547

**Comment by splincode** [comment]
@nsbarsukov @vladimirpotekhin could you review, please?

**Pull request closed without merging**